File: Domain/Models/Pacientes.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Models
{
    public class Pacientes
    {
        public int IdPaciente { get; set; }
        public string? Nombres { get; set; } 
        public string? Apellidos { get; set; } 
        public DateTime FechaNacimiento { get; set; }
        public string? Sexo { get; set; } 
        public string? DUI { get; set; } 
        public string? Telefono { get; set; }
        public string? Correo { get; set; } 
        public string? Direccion { get; set; } 
        public DateTime FechaRegistro { get; set; }
    }
}

File: Frontend/medical-appointments/src/app/services/pacientes.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Paciente {
  idPaciente: number;
  nombres: string;
  apellidos: string;
  fechaNacimiento: string;
  sexo: string;
  dui: string;
  telefono: string;
  correo: string;
  direccion: string;
  fechaRegistro: string;
}

@Injectable({
  providedIn: 'root'
})
export class PacientesService {

  private apiUrl = 'https://localhost:7250/Pacientes';

  constructor(private http: HttpClient) {}

  getPacientes(): Observable<Paciente[]> {
    return this.http.get<Paciente[]>(this.apiUrl);
  }
}